File: src/domain/entities/SupportTicket.ts
export interface SupportTicket {
  id: string
  kind: 'SUGERENCIA' | 'PROBLEMA'
  message: string
  status: 'ABIERTO' | 'RESUELTO'
  createdById: string
  createdByName: string
  createdAt: Date
  response: string
  resolvedAt: Date | null
  resolvedById: string
  resolvedByName: string
}

export function supportTicketKindLabel(kind: SupportTicket['kind']): string {
  return kind === 'PROBLEMA' ? 'Problema' : 'Sugerencia'
}

export function supportTicketStatusLabel(
  status: SupportTicket['status'],
): string {
  return status === 'RESUELTO' ? 'Resuelto' : 'Abierto'
}

File: consorcio/lib/domain/entities/support_ticket.dart
class SupportTicket {
  const SupportTicket({
    required this.id,
    required this.kind,
    required this.message,
    required this.status,
    required this.createdById,
    required this.createdByName,
    required this.createdAt,
    this.response = '',
    this.resolvedAt,
    this.resolvedById = '',
    this.resolvedByName = '',
  });

  final String id;
  final String kind;
  final String message;
  final String status;
  final String createdById;
  final String createdByName;
  final DateTime createdAt;
  final String response;
  final DateTime? resolvedAt;
  final String resolvedById;
  final String resolvedByName;

  bool get isOpen => status != 'RESUELTO';
  bool get isProblem => kind == 'PROBLEMA';

  String get kindLabel => isProblem ? 'Problema' : 'Sugerencia';
  String get statusLabel => isOpen ? 'Abierto' : 'Resuelto';
}
